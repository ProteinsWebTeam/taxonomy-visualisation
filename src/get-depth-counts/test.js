import getDepthCounts from '.';

test('getDepthCount to return correct values', () => {
  const d0 = { depth: 0 };
  const d1 = { depth: 1 };
  const d2 = { depth: 2 };
  const d3 = { depth: 3 };
  const nodeSets = [
    [d0, d1, d1, d2, d2, d2, d3, d3, d3, d3],
    [d0, d1, d1, d2, d2, d2, d3],
    [d0, d1, d2, d3],
    [d0],
    [d0, d1, d1, d2, d2, d2, d2, d2, d3],
  ];
  for (const nodeSet of nodeSets) {
    expect(getDepthCounts(nodeSet)).toMatchSnapshot();
  }
});
