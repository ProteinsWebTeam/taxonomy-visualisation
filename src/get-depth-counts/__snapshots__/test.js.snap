// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDepthCount to return correct values 1`] = `
Array [
  1,
  2,
  3,
  4,
]
`;

exports[`getDepthCount to return correct values 2`] = `
Array [
  1,
  2,
  3,
  1,
]
`;

exports[`getDepthCount to return correct values 3`] = `
Array [
  1,
  1,
  1,
  1,
]
`;

exports[`getDepthCount to return correct values 4`] = `
Array [
  1,
]
`;

exports[`getDepthCount to return correct values 5`] = `
Array [
  1,
  2,
  5,
  1,
]
`;
